app/app/main.py

```diff
@@ -80,55 +84,127 @@ def validate_extension(filename):
     )
  
 
+def remove_folder(path,tdate):
+
+        if not shutil.rmtree(path):
+                logger.info(f"---{tdate}---{path} folder is removed successfully")
+        else:
+                logger.info(f"---{tdate}---Unable to delete the {path}")
+
+
+def remove_file(path,tdate):
+
+           if not os.remove(path):         
+                logger.info(f"---{tdate}---{path} file is removed successfully")
+           else:           
+                logger.info(f"---{tdate}---Unable to delete the {path}")
+
 
+def get_file_or_folder_age(path):
+
+        # getting modified time of the file/folder
+        # time will be in seconds
+        mtime = os.stat(path).st_mtime
+
+        return mtime
+        
 @app.on_event("startup")
 async def startup():
-    await database.connect()
+        await database.connect()
+
+@app.on_event("startup")
+@repeat_every(seconds= 24 * 60 * 60) 
+async def remove_old_files():
+
+
+    if(env('ENABLE_PRUNER')=='true'):
+        today = date.today()
+        logger.info(f"---------Pruner logs for {today}-------")
+
+        config=cfg.get_config()
+        logger.info(config)
+        dirs=config['pruner']
+        for i in dirs:
+
+            folder=config['pruner'][i]['path']
+            days = config['pruner'][i]['days_to_live']
+            path = os.path.join(RESULTS_DIR, folder)
+                        
+            # time.time() returns current time in seconds
+            seconds = time.time() - (days * 24 * 60 * 60)
+
+
+            # checking whether the file is present in path or not
+            if os.path.exists(path):
+
+                    # iterating over each and every folder and file in the path
+                    for root_folder, folders, files in os.walk(path):
+
+                            # checking folder from the root_folder
+                            for folder in folders:
+
+                                    folder_path = os.path.join(root_folder, folder)
+
+                                    if seconds >= get_file_or_folder_age(folder_path):
+
+                                            remove_folder(folder_path,today)
+
+
+                            # checking the current directory files
+                            for file in files:
+
+                                    file_path = os.path.join(root_folder, file)
+
+                                    if seconds >= get_file_or_folder_age(file_path):
+
+                                            remove_file(file_path,today)
+
+
 
 
 @app.on_event("shutdown")
 async def shutdown():
-    await database.disconnect()
+        await database.disconnect()
```

Comment:
The extra indent here and in the rest of the source code should be removed.

Reply:
is this necessary ? i just changed from tabs to spaces , coz tabs were giving me trouble shifting from vscode to vim and vice versa

Reply:
Yes, everything in the code base should have the same indentation pattern of spaces. [VSCode has a remote ssh session](https://code.visualstudio.com/docs/remote/ssh) that can remove much your need for vim. I use it, and I'd be happy to show you how I got going with it.

Your other alternative would be to enable [vim's expandtab option](https://vim.fandom.com/wiki/Converting_tabs_to_spaces). I do not have any familiarity with this, though.

Reply:
right